spec/rubocop/cop/style/multiline_when_then_spec.rb

```diff
@@ -84,4 +84,21 @@
       end
     RUBY
   end
+
+  it 'autocorrects when the body of `when` branch starts ' \
+     'with `then`' do
+    new_source = autocorrect_source(<<~RUBY)
+      case foo
+      when bar
+        then do_something
+      end
+    RUBY
+
+    expect(new_source).to eq(<<~RUBY)
+      case foo
+      when bar
+       do_something
```

Comment:
📝 There is a slight difference in indentation using only this cop. That indentation is corrected using `Layout/IndentationWidth` cop.